### DISCUSSION TITLE
How to solve declaration for object schema with itself as value at some attribute

### DISCUSSION THREAD

**Discussion by ivan-mihalic** [open-circle/valibot#236] [Q&A] [answered]
I am trying to declare object schema `entity` with some attributes like string, number, etc... and for example `children` that is an array of entities. I am trying to do it like this:

```
const EntitySchema = obect({
  id: number(),
  name: string(),
 children: array(EntitySchema)
});
```

I am getting TS error `Block-scoped variable 'EntitySchema' used before its declaration`.
Is there any way how to achieve this?
thank you

**Comment by fabian-hiller** [comment, marked as answer]
This is a recursive schema. Have a look at: #72 and #98

**Reply by ivan-mihalic** [reply]
nice!
thank you very much @fabian-hiller
